dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavResult.java

```diff
@@ -0,0 +1,374 @@
+        if (parent == null)
+            return null; // no parent! I'm the root folder
+        if (parent.equals(FolderAPI.SYSTEM_FOLDER))
+            return "/";
+        Folder folder = APILocator.getFolderAPI()
+            .find(parent, sysuser, true);
+        Identifier ident = APILocator.getIdentifierAPI()
+            .find(folder);
+        return ident.getURI();
+    }
+
+    public NavResult getParent() throws DotDataException, DotSecurityException {
+        String path = getParentPath();
+        if (path != null) {
+            NavResult lazyMe = new NavTool().getNav(APILocator.getHostAPI()
+                .find(hostId, sysuser, true), path, languageId, sysuser);
+
+            return lazyMe;
+        } else
+            return null;
+    }
+
+    public String toString() {
+        if (!isCodeLink()) {
+            String titleToShow;
+            try {
+                titleToShow = getTitle();
+            } catch (Exception e) {
+                titleToShow = title;
+            }
+            return "<a href='" + getHref() + "' title='" + titleToShow + "'>" + titleToShow + "</a>";
+        } else {
+            return getCodeLink();
+        }
+    }
+
+    public List<String> getChildrenFolderIds() {
+        return childrenFolderIds != null ? childrenFolderIds : new ArrayList<String>();
+    }
+
+    public void setChildrenFolderIds(List<String> childrenFolderIds) {
+        this.childrenFolderIds = childrenFolderIds;
+    }
+
+    @Override
+    public Iterator<NavResult> iterator() {
+        try {
+            List<? extends NavResult>  childs= getChildren();
+            List<NavResult> ret = new ArrayList<>();
+            for(NavResult child  : childs) {
+                ret.add(child);
+            }
+            return ret.iterator();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    public String getTarget() {
+        if (type.equals("link"))
+            return target;
+        else
+            return "_self";
+    }
+
+    public void setTarget(String target) {
+
+        this.target = target;
+    }
+
+    public String getPermissionId() {
+        return permissionId;
+    }
+
+    public void setPermissionId(String permissionId) {
+        this.permissionId = permissionId;
+    }
+
+
+    /// Permissionable methods ///
+
+    @Override
+    public String getOwner() {
+        return sysuser.getUserId();
+    }
+
+    @Override
+    public void setOwner(String owner) {}
+
+    @Override
+    public List<PermissionSummary> acceptedPermissions() {
+        return Arrays.asList(new PermissionSummary[] {new PermissionSummary("READ", "READ", PermissionAPI.PERMISSION_READ)});
+    }
+
+    @Override
+    public List<RelatedPermissionableGroup> permissionDependencies(int requiredPermission) {
+        return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1168)